src/routing.js

```diff
@@ -79,25 +71,52 @@ export const parsePath = (history, pathname, search = '') => {
     }
   }
 
-  const [, instanceId, ...appParts] = parts
+  const [, instanceId, ...instancePathParts] = parts
+
+  // The local path of an app (internal or external)
+  const instancePath = `/${
+    instancePathParts ? instancePathParts.join('/') : ''
+  }`
 
   const completeLocator = {
-    path,
+    ...base,
     mode: APP_MODE_ORG,
     dao,
     instanceId: instanceId || 'home',
     params,
-    parts: appParts,
+    instancePath,
   }
 
   return completeLocator
 }
 
+function encodePath(path) {
+  return path
+    .split('/')
+    .map(v => encodeURIComponent(v))
+    .join('/')
+}
+
 // Return a path string for an app instance
-export const getAppPath = ({ dao, instanceId = 'home', params } = {}) => {
-  const paramsPart = params ? `?p=${encodeURIComponent(params)}` : ``
```

Comment:
I guess we'll modify how this is done in permissions, so we don't use the `?p=`?

Does it also work for importing custom labels?

Reply:
`?p` params have been removed since then.